localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_request.py

```diff
@@ -77,13 +79,20 @@ def __call__(
         if not (integration_method := integration["httpMethod"]) or integration_method == "ANY":
             # otherwise, fallback to the request's method
             integration_method = context.invocation_request["method"]
+        # AWS doesn't send a body for these methods. Mock needs the body to get the status code
+        # TODO this might need to be done in the integrations?
+        if (
+            integration_method in [HTTPMethod.GET, HTTPMethod.HEAD, HTTPMethod.OPTIONS]
+            and not integration_type == IntegrationType.MOCK
+        ):
+            body = b""
```

Comment:
@bentsku I added this section after validating it's behavior against aws.

Reply:
Ah, I answered here too 😄 https://github.com/localstack/localstack/pull/11138#discussion_r1664991116